### PR TITLE
Convert CustomBuildScanEnhancements to a normal class to improve consistency

### PR DESCRIPTION
**Pull request by jthurne**
- Pass context as constructor parameters, to avoid having to pass different method parameters.
- Extracted non-static `sysProperty` `projectProperty` and `envVariable` methods to avoid
  differences in method parameters for most calls.
- Reordered methods for consistency

### CODE REVIEW

**Comment by jthurne** [comment]
@etiennestuder do we want to apply the same sort of refactoring to CustomGradleEnterpriseConfig and SystemPropertyOverrides? While it isn't necessary for consistency between the Gradle plugin and Maven extension, it would make the code within a single project more consistent.

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java
Comment (runningcode) [inline common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java:41]: these were previously all static methods because of an issue being compatible with the configuration cache. I think this should just work now that they are members on the class but we should check before merging.

**Review by etiennestuder — APPROVED** [review]
Looking great.

**Comment by etiennestuder** [comment]
> @etiennestuder do we want to apply the same sort of refactoring to CustomGradleEnterpriseConfig and SystemPropertyOverrides? While it isn't necessary for consistency between the Gradle plugin and Maven extension, it would make the code within a single project more consistent.

@jthurne I would merge this PR once you have ensured config caching is still supported (as requested by Nelson). And then create a separate PR to adjust the other Config classes. This makes it a bit easier to grasp the changes.

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java
Comment (jthurne) [reply common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java:41]: Good suggestion. It looks like these changes do cause a problem with the configuration cache. I'm exploring options.

Comment (jthurne) [reply common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java:41]: Configuration cache fix has been added to the PR.

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java
Comment (runningcode) [inline common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java:233]: ah i see, writing it in this way ensure that it doesn't capture the outer class and ensure it is compatible with the configuration cache. nice find. i wonder if we should document this in the code to avoid future incompatible changes or write a test for it.

**Review by runningcode — APPROVED** [review]
looks good to me!

File: common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java
Comment (etiennestuder) [reply common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java:233]: Yes, that was the plan.

**Review by etiennestuder — COMMENTED** [review]
Nice outcome.

**Pull request merged**
